lib/src/model/model_element.dart

```diff
@@ -401,66 +371,17 @@ abstract class ModelElement extends Canonicalization
 
-  List<String> get annotations => annotationsFromMetadata(element.metadata);
-
-  /// Returns linked annotations from a given metadata set, with escaping.
-  // TODO(srawlins): Attempt to revive constructor arguments in an annotation,
-  // akin to source_gen's Reviver, in order to link to inner components. For
-  // example, in `@Foo(const Bar(), baz: <Baz>[Baz.one, Baz.two])`, link to
-  // `Foo`, `Bar`, `Baz`, `Baz.one`, and `Baz.two`.
-  List<String> annotationsFromMetadata(Iterable<ElementAnnotation> md) {
-    var annotationStrings = <String>[];
-    if (md == null) return annotationStrings;
-    for (var a in md) {
-      var annotation = (const HtmlEscape()).convert(a.toSource());
-      var annotationElement = a.element;
-
-      if (annotationElement is ConstructorElement) {
-        // TODO(srawlins): I think we should actually link to the constructor,
-        // which may have details about parameters. For example, given the
-        // annotation `@Immutable('text')`, the constructor documents what the
-        // parameter is, and the class only references `immutable`. It's a
-        // lose-lose cycle of mis-direction.
-        annotationElement =
-            (annotationElement as ConstructorElement).returnType.element;
-      } else if (annotationElement is PropertyAccessorElement) {
-        annotationElement =
-            (annotationElement as PropertyAccessorElement).variable;
-      }
-      if (annotationElement is Member) {
-        annotationElement = (annotationElement as Member).declaration;
-      }
-
-      // Some annotations are intended to be invisible (such as `@pragma`).
-      if (!_shouldDisplayAnnotation(annotationElement)) continue;
-
-      var annotationModelElement =
-          packageGraph.findCanonicalModelElementFor(annotationElement);
-      if (annotationModelElement != null) {
-        annotation = annotation.replaceFirst(
-            annotationModelElement.name, annotationModelElement.linkedName);
-      }
-      annotationStrings.add(annotation);
-    }
-    return annotationStrings;
-  }
-
-  bool _shouldDisplayAnnotation(Element annotationElement) {
-    if (annotationElement is ClassElement) {
-      var annotationClass =
-          packageGraph.findCanonicalModelElementFor(annotationElement) as Class;
-      if (annotationClass == null && annotationElement != null) {
-        annotationClass =
-            ModelElement.fromElement(annotationElement, packageGraph) as Class;
-      }
-
-      return annotationClass == null ||
-          packageGraph.isAnnotationVisible(annotationClass);
-    }
-    // We cannot resolve it, which does not prevent it from being displayed.
-    return true;
-  }
-
+  Iterable<Annotation> _annotations;
+  // Skips over annotations with null elements or that are otherwise
+  // supposed to be invisible (@pragma).  While technically they are
+  // invalid code from analyzer's perspective they are present in sky_engine
```

Comment:
I'm not sure how to read this sentence. An annotation like `@pragma` is invalid? Does invalid mean analyzer crashes or something?

Reply:
This got lost in multiple rewrites; the only indicator of invalid code is the null element.  Updated.